openmpi: 4.0.5 -> 4.1.0

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from https://repology.org/metapackage/openmpi/versions.


meta.description for openmpi is: "Open source MPI-3 implementation"

meta.homepage for openmpi is: "https://www.open-mpi.org/"

meta.changelog for openmpi is: ""


###### Updates performed
- Version update

###### To inspect upstream changes





###### Impact

<details>
<summary>
<b>Checks done</b> (click to expand)
</summary>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed

- 0 of 0 passed binary check by having a zero exit code.
- 0 of 0 passed binary check by having the new version present in output.
- found 4.1.0 with grep in /nix/store/r48gq71w0kz04phjlqwzz7i28n1zpd91-openmpi-4.1.0
- directory tree listing: https://gist.github.com/a5d0ad6012de83cd7132c08169b5f561

- du listing: https://gist.github.com/3083815e46ee30383c75379f7c9784f5

---

</details>
<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
420 total rebuild path(s)

131 package rebuild(s)

116 x86_64-linux rebuild(s)
98 i686-linux rebuild(s)
92 x86_64-darwin rebuild(s)
114 aarch64-linux rebuild(s)


First fifty rebuilds by attrpath
cntk
dl-poly-classic-mpi
elmerfem
freecad
getdp
globalarrays
gromacsDoubleMpi
gromacsMpi
haskellPackages.haskell-mpi
hdf5-mpi
highfive-mpi
hpcg
hpl
ior
lammps-mpi
migrate
netcdf-mpi
neuron-full
neuron-mpi
openmolcas
openmpi
paraview
parmetis
precice
python37Packages.arviz
python37Packages.baselines
python37Packages.cntk
python37Packages.dask-gateway
python37Packages.dask-jobqueue
python37Packages.dask-ml
python37Packages.dask-mpi
python37Packages.dask-xgboost
python37Packages.datashader
python37Packages.dftfit
python37Packages.distributed
python37Packages.dm-sonnet
python37Packages.edward
python37Packages.fenics
python37Packages.fipy
python37Packages.graph_nets
python37Packages.h5py-mpi
python37Packages.lammps-cython
python37Packages.mask-rcnn
python37Packages.mpi4py
python37Packages.neuron-mpi
python37Packages.optuna
python37Packages.pymc3
python37Packages.rl-coach
python37Packages.streamz
python37Packages.stumpy
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from Cachix**:
```
nix-store -r /nix/store/r48gq71w0kz04phjlqwzz7i28n1zpd91-openmpi-4.1.0 \
  --option binary-caches 'https://cache.nixos.org/ https://nix-community.cachix.org/' \
  --option trusted-public-keys '
  nix-community.cachix.org-1:mB9FSh9qf2dCimDSUo8Zy7bkq5CX+/rkCWyvRCYg3Fs=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The Cachix cache is only trusted for this store-path realization.)
For the Cachix download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A openmpi https://github.com/r-ryantm/nixpkgs/archive/a6d89372fda9faae7d503b896783926005c46935.tar.gz
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/r48gq71w0kz04phjlqwzz7i28n1zpd91-openmpi-4.1.0
ls -la /nix/store/r48gq71w0kz04phjlqwzz7i28n1zpd91-openmpi-4.1.0/bin
```

---

</details>
<br/>



### Pre-merge build results

NixPkgs review skipped

---

###### Maintainer pings

cc @markuskowa for testing.

r-ryantm committed:
openmpi: 4.0.5 -> 4.1.0

markuskowa commented:
@GrahamcOfBorg build scalapack

markuskowa commented:
The openmpi update breaks at least `mpi4py` (the test phase does not run properly in the sandbox anymore). Let's hold back the merge until the problem is solved.

iblech committed:
whitebophir: 1.7.0 -> 1.8.0

iblech committed:
nixos/whitebophir: expose new option provided by whitebophir 1.8.0

iblech committed:
whitebophir: 1.8.0 -> 1.9.0

ajs124 committed:
rdma-core: python2 -> python3

vbgl committed:
ocamlPackages.core_kernel: 0.14.0 → 0.14.1

iblech committed:
whitebophir: 1.9.0 -> 1.9.1

Ma27 committed:
grafana: 7.4.1 -> 7.4.3

ChangeLogs:
   * https://github.com/grafana/grafana/releases/tag/v7.4.2
   * https://github.com/grafana/grafana/releases/tag/v7.4.3

Ma27 committed:
grafana-image-renderer: 2020-12-01 -> 2.0.1

ChangeLog: https://github.com/grafana/grafana-image-renderer/releases/tag/v2.0.1

saschagrunert committed:
ginkgo: 1.15.0 -> 1.15.1

Signed-off-by: Sascha Grunert <sgrunert@redhat.com>

siraben committed:
csa: expand platforms to unix

siraben committed:
ceptre: expand platforms to unix

siraben committed:
duktape: fix darwin build

siraben committed:
brigand: expand platforms to all

siraben committed:
backward-cpp: expand platforms to all

siraben committed:
ethash: expand platforms to unix

siraben committed:
gensio: expand platforms to unix

siraben committed:
rman: fix darwin build

siraben committed:
wipe: expand platforms to all

siraben committed:
hashcat-utils: expand platforms to unix

siraben committed:
lyra: expand platforms to unix

siraben committed:
openpa: expand platforms to unix

siraben committed:
croaring: expand platforms to all

siraben committed:
cminpack: fix darwin build

siraben committed:
uptimed: expand platforms to unix

siraben committed:
ripemime: fix darwin build

siraben committed:
jikespg: fix darwin build

rnhmjoj committed:
nixos/privoxy: document repeated settings

r-ryantm committed:
slurm: 20.11.3.1 -> 20.11.4.1

wizeman committed:
go-task: 2.8.1 -> 3.2.2

06kellyjac committed:
open-policy-agent: 0.26.0 -> 0.27.0

Set buildFlagsArray using bash
use -s and -w to minify the result
Add installCheckPhase
Add @06kellyjac as a maintainer

06kellyjac committed:
terragrunt: 0.28.7 -> 0.28.8

terlar committed:
emacs: do not use project package from elpa for emacs >= 28

The `project` package bundled with Emacs 28 is newer than the one
provided by GNU ELPA. This may or may not change in the future.

For now it is definitely the case and since the package is under heavy
development a lot of features are missing in the one on GNU ELPA which in
turn is newer than the one that comes bundled with Emacs 27.

06kellyjac committed:
nerdctl: 0.6.1 -> 0.7.0

Set buildFlagsArray using bash
Completions have been added but the command will fail without access to
a containerd socket

SuperSandro2000 committed:
Merge pull request #115578 from wizeman/u/upd-go-task

go-task: 2.8.1 -> 3.2.2

rnhmjoj committed:
Merge pull request #115572 from rnhmjoj/privoxy-doc

nixos/privoxy: document repeated settings

markuskowa committed:
Merge pull request #114918 from helsinki-systems/rdma-core-python3

rdma-core: python2 -> python3

markuskowa committed:
Merge pull request #115563 from r-ryantm/auto-update/slurm

slurm: 20.11.3.1 -> 20.11.4.1

adisbladis committed:
Merge pull request #115592 from terlar/emacs28-project

emacs: do not use project package from elpa for emacs >= 28

SuperSandro2000 committed:
Merge pull request #115589 from 06kellyjac/nerdctl

nerdctl: 0.6.1 -> 0.7.0

SuperSandro2000 committed:
Merge pull request #115585 from 06kellyjac/open-policy-agent

open-policy-agent: 0.26.0 -> 0.27.0

SuperSandro2000 committed:
Merge pull request #115586 from 06kellyjac/terragrunt

terragrunt: 0.28.7 -> 0.28.8

SuperSandro2000 committed:
Merge pull request #115424 from siraben/darwin-mass-fix-buildInputs=0

ProofOfKeags committed:
lnd: 0.12.0 -> 0.12.1

primeos committed:
chromiumDev: 90.0.4430.11 -> 90.0.4430.19

SuperSandro2000 committed:
Merge pull request #115392 from saschagrunert/ginkgo

ginkgo: 1.15.0 -> 1.15.1

SuperSandro2000 committed:
Merge pull request #112689 from iblech/patch-whitebophir180

Ma27 committed:
Merge pull request #115366 from Ma27/grafana-updates

grafana*: package updates

prusnak committed:
Merge pull request #115610 from ProofOfKeags/lnd-0.12.1

lnd: 0.12.0 -> 0.12.1

markuskowa committed:
pythonPackage.mpi4py: add patch for openmpi 4.1.0

markuskowa commented:
rebase gone wrong. I will open this as new PR.
